src/EventStore.Projections.Core/Services/Management/ProjectionManager.cs

```diff
@@ -797,128 +851,308 @@ private readonly
 		}
 
-		private void PostNewProjection(ProjectionManagementMessage.Command.Post message, IEnvelope replyEnvelope) {
+		private void ReadProjectionPossibleStream(
+			string projectionName,
+			Action<ClientMessage.ReadStreamEventsBackwardCompleted> onComplete) {
 			var corrId = Guid.NewGuid();
 			_readDispatcher.Publish(
 				new ClientMessage.ReadStreamEventsBackward(
 					corrId,
 					corrId,
 					_readDispatcher.Envelope,
-					ProjectionNamesBuilder.ProjectionsStreamPrefix + message.Name,
+					ProjectionNamesBuilder.ProjectionsStreamPrefix + projectionName,
 					0,
 					_readEventsBatchSize,
 					resolveLinkTos: false,
 					requireMaster: false,
 					validationStreamVersion: null,
 					user: SystemAccount.Principal),
-				m => CompletedReadingPossibleStream(m, message, replyEnvelope));
+				onComplete);
+		}
+
+		private void ReadProjectionPossibleStreamCompleted(
+			ClientMessage.ReadStreamEventsBackwardCompleted completed,
+			NewProjectionInitializer initializer,
+			IEnvelope replyEnvelope) {
+
+			long version = -1;
+			if (completed.Result == ReadStreamResult.Success) {
+				version = completed.LastEventNumber + 1;
+			}
+
+			InitializeNewProjection(initializer, version, replyEnvelope);
+		}
+
+		private void OnProjectionsRegistrationCaughtUp() {
+			_logger.Debug($"PROJECTIONS: Caught up with projections registration. Next expected version: {_projectionsRegistrationExpectedVersion}");
+		}
+
+		private void BeginWriteProjectionDeleted(
+			ClientMessage.WriteEvents writeDelete,
+			ProjectionManagementMessage.Internal.Deleted message,
+			Action<long> onCompleted,
+			int retryCount = ProjectionCreationRetryCount) {
+			_writeDispatcher.Publish(
+				writeDelete,
+				m => WriteProjectionDeletedCompleted(m, writeDelete, message, onCompleted, retryCount));
+		}
+
+		private void WriteProjectionDeletedCompleted(ClientMessage.WriteEventsCompleted writeCompleted,
+			ClientMessage.WriteEvents writeDelete,
+			ProjectionManagementMessage.Internal.Deleted message,
+			Action<long> onCompleted,
+			int retryCount) {
+
+			if (writeCompleted.Result == OperationResult.Success) {
+				onCompleted?.Invoke(writeCompleted.LastEventNumber);
+				return;
+			}
+
+			_logger.Info(
+				"PROJECTIONS: Projection '{projection}' deletion has not been written to {stream}. Error: {e}",
+				message.Name,
+				ProjectionNamesBuilder.ProjectionsRegistrationStream,
+				Enum.GetName(typeof(OperationResult), writeCompleted.Result));
+
+			if (writeCompleted.Result == OperationResult.CommitTimeout || writeCompleted.Result == OperationResult.ForwardTimeout
+																	   || writeCompleted.Result == OperationResult.PrepareTimeout) {
+				if (retryCount > 0) {
+					_logger.Info("PROJECTIONS: Retrying write projection deletion for {projection}", message.Name);
+					BeginWriteProjectionDeleted(writeDelete, message, onCompleted, retryCount - 1);
+					return;
+				}
+			}
+
+			if (writeCompleted.Result == OperationResult.WrongExpectedVersion
```

Comment:
While we can change this for writes for the projection being created, the projection deletes can be triggered internally. This happens when a query has finished running, or when a OneTime projection is disposed.

Because there are no clients, and these deletions can happen asynchronously, we need to retry in the case of WrongExpected versions.